apps/sasquatch/src/main/java/com/microsoft/appcenter/sasquatch/activities/SettingsActivity.java

```diff
@@ -570,6 +570,9 @@ public void onReset() {
 
                 @Override
                 public void onSave(String value) {
+                    if (value != null && value.isEmpty()){
```

Comment:
It is by design to test empty string passed to SDK and see that AppCenter accepts it and sends it and that One Collector rejects it with `E/AppCenter: userId must not be empty.`.

You have to use the Reset button to erase.

@jaelim-ms we should cancel this bug.

Reply:
This is for child targets which we don't have a dialog and reset button. We can implement this one for it but it could be tested by the one in the root level.

Reply:
Ok this is the wrong fix then and needs to be started over.
After talking to @jaelim-ms what we need to modify is this UI:

<img width="309" alt="Screen Shot 2019-06-18 at 1 56 42 PM" src="https://user-images.githubusercontent.com/623672/59719297-ec962a00-91d0-11e9-8968-ee639e65b597.png">

As you can see UserId is blank there, we need to say something like "unset" for null and "empty string" for empty string for the displayed value in the settings list.

The editor (onSave) must not be patched.

Reply:
> This is for child targets which we don't have a dialog and reset button. We can implement this one for it but it could be tested by the one in the root level.
The bug was reported at root level, not for children.

Reply:
I update the fix to address the comment. But I still patch the onSave method.